### PR TITLE
feat: upgrade Deno core and extensions

### PR DESCRIPTION
**Pull request by bajtos**
Upgrade Deno crates:
- `deno_core` from `0.171.0` to `0.173.0`
- `deno_runtime` from `0.97.0` to `0.99.0`

Rework our internal modules to use the new ES module style (see https://github.com/denoland/deno/pull/17648 and and https://github.com/denoland/deno/pull/17683)

**The changes are best reviewed with whitespace-changes ignored.**

Rework the way how we call async ops - see https://github.com/denoland/deno/pull/17899

Disable Windows support for the time being because GitHub Action Windows runners don't have enough disk space for building Zinnia 😱 I opened #99 to investigate and fix this problem.

See #73

### CODE REVIEW

**Comment by bajtos** [comment]
> Disable Windows support for the time being because GitHub Action Windows runners don't have enough disk space for building Zinnia 😱

I opened a follow-up issue here: https://github.com/filecoin-station/zinnia/issues/99

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -23,8 +23,8 @@ jobs:
             os: macos-latest
           - target: aarch64-apple-darwin
             os: macos-latest
-          - target: x86_64-pc-windows-msvc
-            os: windows-latest
+          # - target: x86_64-pc-windows-msvc
```
Comment (juliangruber) [inline .github/workflows/ci.yaml:26]:
```suggestion
          # Support Windows: https://github.com/filecoin-station/zinnia/issues/99
          # - target: x86_64-pc-windows-msvc
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -33,12 +33,12 @@ jobs:
             os: macos-latest
             name: zinnia-macos-arm64.zip
 
-          - target: x86_64-pc-windows-msvc
-            os: windows-latest
-            name: zinnia-windows-x64.zip
-            rustflags: -C target-feature=+crt-static -C lto=off
-            #                                        ^^^^^^^^^^
-            # LTO is temporarily disabled, see https://github.com/rust-lang/rust/issues/107781
+          # - target: x86_64-pc-windows-msvc
```
Comment (juliangruber) [inline .github/workflows/release.yml:36]:
```suggestion
          # Support Windows: https://github.com/filecoin-station/zinnia/issues/99
          # - target: x86_64-pc-windows-msvc
```

**Review by juliangruber — APPROVED** [review]
While that probably wasn't the most fun exercise, I'm happy that such a fundamental upgrade wasn't bad for us, and now more confident that this is a good direction

**Pull request merged**